### PR TITLE
Add bitcode argument for compiling with bitcode on iOS

### PR DESCRIPTION
**Pull request by dnfield**
To make this really work, we'll need to land other changes to .S files to include the __LLVM.__asm sections.

Related to https://github.com/flutter/flutter/issues/15288

### CODE REVIEW

**Comment by dnfield** [comment]
Actually, this seems to just work without adding those symbols...  Still verifying.

File: build/toolchain/mac/BUILD.gn [hunk 1]
```diff
@@ -263,10 +271,16 @@ mac_toolchain("ios_clang_x64") {
 mac_toolchain("clang_x64") {
   toolchain_cpu = "x64"
   toolchain_os = "mac"
-  prefix = rebase_path(llvm_bin_path, root_build_dir)
-  ar = "$prefix/llvm-ar"
-  cc = "$prefix/clang"
-  cxx = "$prefix/clang++"
+  if (!enable_bitcode) {
```
Comment (chinmaygarde) [inline build/toolchain/mac/BUILD.gn]: We can avoid this for host builds. `x64`

File: build/toolchain/mac/BUILD.gn [hunk 2]
```diff
@@ -276,10 +290,16 @@ mac_toolchain("clang_x64") {
 mac_toolchain("clang_x86") {
   toolchain_cpu = "i386"
   toolchain_os = "mac"
-  prefix = rebase_path(llvm_bin_path, root_build_dir)
-  ar = "$prefix/llvm-ar"
-  cc = "$prefix/clang"
-  cxx = "$prefix/clang++"
+  if (!enable_bitcode) {
```
Comment (chinmaygarde) [inline build/toolchain/mac/BUILD.gn]: We can avoid this for host builds. `i386`

File: build/toolchain/mac/BUILD.gn (hunk 1 above)
Comment (dnfield) [reply build/toolchain/mac/BUILD.gn]: I get linking errors if I don't do this because of gen_snapshot

File: build/toolchain/mac/BUILD.gn (hunk 2 above)
Comment (dnfield) [reply build/toolchain/mac/BUILD.gn]: Ditto

**Pull request merged**